astrokat/observe_main.py

```diff
@@ -218,17 +225,47 @@ def cadence_target(target_list):
     return False
 
 
-def above_horizon(katpt_target, horizon=20.0, duration=0.0):
+def above_horizon(target,
+                  observer,
+                  horizon=20.0,
+                  duration=0.0):
     """Check target visibility."""
-    [azim, elev] = katpt_target.azel(time.time())
-    if not elev > ephem.degrees(str(horizon)):
+    # use local copies so you do not overwrite target time attribute
+    horizon = ephem.degrees(str(horizon))
+
+    if type(target) is not ephem.FixedBody:
+        # anticipate katpoint special target for AzEl targets
+        if 'alt' not in vars(target):
+            raise RuntimeError('Unknown target type, exiting...')
+        # 'StationaryBody' objects do not have RaDec coordinates
+        # check pointing altitude is above minimum elevation limit
+        if target.alt < horizon:
+            return False
+        else:
+            return True
```

Comment:
Instead of writing it like that, you could rewrite it this way.

```suggestion
        return bool(target.alt < horizon)
```